### PR TITLE
feat(cli): wperf record/report CLI scaffold with signal handling

### PR DESCRIPTION
**Pull request by adrian-mason**
## Summary
- Add clap-based CLI scaffold with `record`, `report`, and `version` subcommands
- Implement `record` subcommand: argument parsing (`-o`, `-d`, `--buffer-size`), SIGINT/SIGTERM graceful shutdown via `signal-hook` `AtomicBool`, and BPF pipeline orchestration seam behind `#[cfg(feature = "bpf")]`
- Runtime error (not compile error) when BPF feature is absent — default build remains compilable
- BPF-enabled build returns explicit `NotYetWired` error until skeleton pipeline is integrated
- Does **not** deliver end-to-end record pipeline (blocked on BPF skeleton build wiring)

## Authoritative Inputs
- final-design.md §1.2 (unified offline CLI model: `wperf record/report/version`)
- final-design.md §4.1-4.4 (wPRF format + crash recovery)
- ADR-002 (feature probing: `probe_all()` → `FeatureMatrix`)
- ADR-004 (transport abstraction: `EventTransport` trait)
- PR #86 (transport abstraction landed)
- PR #90 (Authoritative Inputs / Deviations process)

## Deviations
None

## Dependency Checklist
- [ ] `clap` 4.6 with `derive` feature — new dependency, not in baseline (CLI framework, standard choice)
- [ ] `signal-hook` 0.4 — new dependency, not in baseline (POSIX signal handling)
- [ ] No deviations from existing baseline deps

## Review Checklist
- [x] **Design conformance**: implementation matches the ADRs/specs listed in Authoritative Inputs — *Critic PASS, Challenger PASS*
- [x] **Deviations declared**: any spec drift is explicitly listed above with a doc update reference — *Critic PASS, Challenger PASS*
- [x] **Code correctness**: logic, error handling, edge cases — *Critic PASS (3 blockers fixed: signal flag, false-success, version subcommand)*
- [x] **Tests**: new/changed code has adequate test coverage — *15 new tests (CLI + record)*
- [x] **CI green**: all required checks pass ✅

## Test Plan
- [x] `cargo test --lib -- cli` — 8 CLI parsing tests pass (including duration validation)
- [x] `cargo test --lib -- record` — 7 record module tests pass
- [x] `cargo clippy --all-targets -- -D warnings` — clean
- [x] `cargo fmt --check` — clean
- [x] Full `cargo test` — all pass

### CODE REVIEW

File: src/record.rs [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+        Self::Io(e)
+    }
+}
+
+/// Run the `wperf record` subcommand.
+///
+/// Orchestration flow (once BPF skeleton is wired):
+/// 1. Register SIGINT/SIGTERM handlers → `running` flag
+/// 2. `probe::probe_all()` → `FeatureMatrix`
+/// 3. Open BPF skeleton → configure transport → load → attach
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/record.rs:74]: Fixed in `3dfaf1e` — renamed to `stop_requested`, initialized to `false`, loop checks `!stop_requested.load()`. @gemini-code-assist

File: src/record.rs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+    // Returning an explicit error avoids producing a misleading empty trace.
+    Err(RecordError::NotYetWired)
+}
+
+/// Non-BPF build: runtime error at invocation boundary.
+#[cfg(not(feature = "bpf"))]
+fn record_impl(_args: &RecordArgs, _stop_requested: &Arc<AtomicBool>) -> Result<(), RecordError> {
+    Err(RecordError::NoBpfSupport)
+}
+
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/record.rs:119]: Fixed in `3dfaf1e` — same commit as above, BPF path now also removed (returns `NotYetWired`). @gemini-code-assist

File: src/cli.rs [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+/// Arguments for the `record` subcommand.
+#[derive(clap::Args, Debug)]
+pub struct RecordArgs {
+    /// Output file path (default: wperf.data).
+    #[arg(short, long, default_value = "wperf.data")]
+    pub output: PathBuf,
+
+    /// Recording duration in seconds. If omitted, records until Ctrl+C.
+    #[arg(short, long)]
+    pub duration: Option<f64>,
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/cli.rs:38]: Fixed in `9a55c2e` — added `parse_positive_duration()` custom value parser that rejects zero and negative values. @gemini-code-assist

File: src/cli.rs [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+    /// Transport buffer size in bytes.
+    ///
+    /// For ringbuf transport: sets `max_entries` (must be power of 2).
+    /// For perfarray transport: sets per-CPU buffer size.
+    /// Default: 16 MiB for ringbuf, 1 MiB/CPU for perfarray.
+    #[arg(long)]
+    pub buffer_size: Option<u32>,
```
[the remark this answers is not in the document]
Comment (adrian-mason) [reply src/cli.rs:46]: Acknowledged but deferring power-of-2 validation to record-time rather than CLI parsing. Reason: only ringbuf requires power-of-2; perfarray path uses pages-per-cpu which has different constraints. Validating at transport configuration time is more precise. @gemini-code-assist
